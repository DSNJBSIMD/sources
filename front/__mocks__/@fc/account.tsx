import type { PropsWithChildren } from 'react';
import React from 'react';

export const AccountContext = React.createContext(expect.any(Object));

export const AccountProvider = jest.fn(
  ({ children, config }: { config: unknown } & PropsWithChildren) => (
    <AccountContext.Provider value={config}>
      <div data-mockid="AccountProvider">
        <div>{children}</div>
      </div>
    </AccountContext.Provider>
  ),
);

export const ConnectValidator = {
  validate: jest.fn(),
};

export const Options = {
  CONFIG_NAME: 'Account',
};
