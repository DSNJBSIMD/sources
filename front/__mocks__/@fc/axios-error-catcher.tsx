import type { PropsWithChildren } from 'react';
import React from 'react';

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));

export const AxiosErrorCatcherProvider = jest.fn(
  ({ children, value }: { value: unknown } & PropsWithChildren) => (
    <AxiosErrorCatcherContext.Provider value={value}>{children}</AxiosErrorCatcherContext.Provider>
  ),
);
