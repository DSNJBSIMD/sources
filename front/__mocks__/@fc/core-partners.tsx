export const CreateInstanceButton = jest.fn(() => <div data-mockid="CreateInstanceButton" />);

export const InstanceComponent = jest.fn(() => <div data-mockid="InstanceComponent" />);

export const InstancesListComponent = jest.fn(() => <div data-mockid="InstancesListComponent" />);

export const PartnersService = {
  commit: jest.fn(),
  get: jest.fn(),
};

export enum Environment {
  SANDBOX = 'SANDBOX',
  PRODUCTION = 'PRODUCTION',
}

export enum Options {
  CONFIG_NAME = 'Partners',
}

export enum PublicationStatus {
  DRAFT = 'DRAFT',
  PENDING = 'PENDING',
  PUBLISHED = 'PUBLISHED',
}
