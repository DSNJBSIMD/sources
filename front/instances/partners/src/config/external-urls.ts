export const ExternalUrls: Record<string, string> = {
  spConfigurationDocUrl:
    'https://docs.partenaires.franceconnect.gouv.fr/fs/devenir-fs/projet-bac-a-sable/#configuration-de-votre-fournisseur-de-service',
};
