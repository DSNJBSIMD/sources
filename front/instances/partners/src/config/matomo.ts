import type { MatomoConfig } from '@fc/matomo';

export const Matomo: MatomoConfig = {
  siteId: 309,
  url: 'https://stats.data.gouv.fr/',
};
