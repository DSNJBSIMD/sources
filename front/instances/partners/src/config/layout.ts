import { LogoFranceConnect } from '@fc/assets';
import type { LayoutConfig } from '@fc/layout';

export const Layout: LayoutConfig = {
  features: {
    showLicence: false,
    showServiceTitle: true,
  },
  footer: {
    description:
      'L’Espace partenaires FranceConnect est un dispositif qui permet aux partenaires de gérer leur fournisseur de service FranceConnect.',
    links: [
      {
        external: true,
        href: 'https://www.legifrance.gouv.fr',
        label: 'legifrance.gouv.fr',
        title: 'Accéder au site legifrance.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.info.gouv.fr',
        label: 'info.gouv.fr',
        title: 'Accéder au site info.gouv.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://www.service-public.fr/',
        label: 'service-public.fr',
        title: 'Accéder au site service-public.fr nouvelle fenêtre',
      },
      {
        external: true,
        href: 'https://data.gouv.fr',
        label: 'data.gouv.fr',
        title: 'Accéder au site data.gouv.fr nouvelle fenêtre',
      },
    ],
    navigation: [
      {
        external: false,
        href: '/plan-du-site',
        label: 'Plan du site',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/accessibilite',
        label: 'Accessibilité : non conforme',
      },
      {
        external: false,
        href: '/mentions-legales',
        label: 'Mentions légales',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Données personnelles',
      },
      {
        external: true,
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Gestion des cookies',
      },
    ],
  },
  service: {
    baseline: 'Espace Partenaires',
    homepage: '/',
    logo: LogoFranceConnect,
    name: 'FranceConnect',
  },
  sitemap: [
    {
      href: '/',
      label: 'Accueil',
    },
    {
      href: '/mentions-legales',
      label: 'Mentions Légales',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/licence',
      label: 'Licence',
    },
    {
      external: true,
      href: 'https://franceconnect.gouv.fr/accessibilite',
      label: 'Accessibilité',
    },
  ],
};
