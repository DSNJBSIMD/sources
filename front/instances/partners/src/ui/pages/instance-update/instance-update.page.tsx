import React from 'react';

import { EventTypes, HeadingTag } from '@fc/common';
import { ConfigService } from '@fc/config';
import type { ExternalUrlsInterface } from '@fc/core-partners';
import { Options } from '@fc/core-partners';
import { AlertComponent, LinkComponent, Sizes } from '@fc/dsfr';
import { DTO2FormComponent } from '@fc/dto2form';
import { t } from '@fc/i18n';

import { useInstanceUpdate } from '../../../hooks';

export const InstanceUpdatePage = React.memo(() => {
  const { initialValues, schema, submitHandler, title } = useInstanceUpdate();

  const { spConfigurationDocUrl } = ConfigService.get<ExternalUrlsInterface>(
    Options.CONFIG_EXTERNAL_URLS,
  );

  return (
    <React.Fragment>
      <h1 className="fr-col-12 fr-col-md-8">{t('Partners.updatepage.title')}</h1>
      <AlertComponent
        className="fr-col-12 fr-col-md-8 fr-mb-4w"
        size={Sizes.MEDIUM}
        title={t('Partners.instance.noticeTitle')}
        type={EventTypes.INFO}>
        <p>
          Consulter notre{' '}
          <LinkComponent
            dataTestId="documentation-partners-link"
            href={spConfigurationDocUrl}
            label="documentation partenaires"
            rel="noopener noreferrer external"
            target="_blank"
          />
        </p>
      </AlertComponent>
      <div className="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full">
        <div className="fr-col-offset-1 fr-col-10">
          <DTO2FormComponent
            config={{ id: 'DTO2Form-instance-update', title, titleHeading: HeadingTag.H2 }}
            initialValues={initialValues}
            schema={schema}
            submitLabel={t('Form.submit.label')}
            onSubmit={submitHandler}
          />
        </div>
      </div>
    </React.Fragment>
  );
});

InstanceUpdatePage.displayName = 'InstanceUpdatePage';
