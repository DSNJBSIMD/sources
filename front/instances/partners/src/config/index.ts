import { Account } from './account';
import { DTO2Form } from './dto2form';
import { ExternalUrls } from './external-urls';
import { HttpClient } from './http-client';
import { Layout } from './layout';
import { Matomo } from './matomo';
import { Partners } from './partners';

export const AppConfig = {
  Account,
  DTO2Form,
  ExternalUrls,
  HttpClient,
  Layout,
  Matomo,
  Partners,
};
