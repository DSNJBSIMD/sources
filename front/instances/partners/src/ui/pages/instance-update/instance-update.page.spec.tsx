import { render } from '@testing-library/react';

import { type AnyObjectInterface, HeadingTag } from '@fc/common';
import { ConfigService } from '@fc/config';
import { AlertComponent, LinkComponent } from '@fc/dsfr';
import type { SchemaFieldType } from '@fc/dto2form';
import { DTO2FormComponent } from '@fc/dto2form';
import { t } from '@fc/i18n';

import { useInstanceUpdate } from '../../../hooks';
import { InstanceUpdatePage } from './instance-update.page';

jest.mock('../../../hooks/instance-update/instance-update.hook');

describe('InstanceUpdatePage', () => {
  const submitHandlerMock = jest.fn();
  const initialValuesMock = Symbol('any-initial-values-mock') as unknown as AnyObjectInterface;
  const schemaMock = Symbol('any-schema-mock') as unknown as SchemaFieldType[];
  const configMock = {
    spConfigurationDocUrl: Symbol('any-spConfigurationDocUrl-mock') as unknown as string,
  };

  beforeEach(() => {
    jest.mocked(useInstanceUpdate).mockReturnValueOnce({
      initialValues: initialValuesMock,
      schema: schemaMock,
      submitHandler: submitHandlerMock,
      title: 'any-instance-name-mock',
    });
    jest.mocked(ConfigService.get).mockReturnValueOnce(configMock);
    jest
      .mocked(t)
      .mockReturnValueOnce('Partners-form-updateTitle-mock')
      .mockReturnValueOnce('Partners.instance.noticeTitle-mock')
      .mockReturnValueOnce('Form.submit.label-mock');
  });

  it('should match the snapshot', () => {
    // When
    const { container, getByText } = render(<InstanceUpdatePage />);
    const titleElt = getByText('Partners-form-updateTitle-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(titleElt).toBeInTheDocument();
    expect(t).toHaveBeenCalledTimes(3);
    expect(t).toHaveBeenNthCalledWith(1, 'Partners.updatepage.title');
    expect(t).toHaveBeenNthCalledWith(2, 'Partners.instance.noticeTitle');
    expect(t).toHaveBeenNthCalledWith(3, 'Form.submit.label');
    expect(useInstanceUpdate).toHaveBeenCalledOnce();
    expect(useInstanceUpdate).toHaveBeenCalledWith();
    expect(DTO2FormComponent).toHaveBeenCalledOnce();
    expect(DTO2FormComponent).toHaveBeenCalledWith(
      {
        config: {
          id: 'DTO2Form-instance-update',
          title: 'any-instance-name-mock',
          titleHeading: HeadingTag.H2,
        },
        initialValues: initialValuesMock,
        onSubmit: submitHandlerMock,
        schema: schemaMock,
        submitLabel: 'Form.submit.label-mock',
      },
      undefined,
    );
  });

  it('should call AlertComponent with info params', () => {
    // When
    const { container } = render(<InstanceUpdatePage />);

    // Then
    expect(container).toMatchSnapshot();
    expect(AlertComponent).toHaveBeenCalledOnce();
    expect(AlertComponent).toHaveBeenCalledWith(
      {
        children: expect.any(Object),
        className: 'fr-col-12 fr-col-md-8 fr-mb-4w',
        size: 'md',
        title: 'Partners.instance.noticeTitle-mock',
        type: 'info',
      },
      undefined,
    );
  });

  it('should call LinkComponent with params', () => {
    // When
    const { container } = render(<InstanceUpdatePage />);

    // Then
    expect(container).toMatchSnapshot();
    expect(LinkComponent).toHaveBeenCalledOnce();
    expect(LinkComponent).toHaveBeenCalledWith(
      {
        dataTestId: 'documentation-partners-link',
        href: configMock.spConfigurationDocUrl,
        label: 'documentation partenaires',
        rel: 'noopener noreferrer external',
        target: '_blank',
      },
      undefined,
    );
  });
});
