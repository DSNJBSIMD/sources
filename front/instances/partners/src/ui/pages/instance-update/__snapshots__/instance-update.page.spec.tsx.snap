// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceUpdatePage should call AlertComponent with info params 1`] = `
<div>
  <h1
    class="fr-col-12 fr-col-md-8"
  >
    Partners-form-updateTitle-mock
  </h1>
  <div
    data-mockid="AlertComponent"
  >
    <div>
      AlertComponent
    </div>
    <div>
      <p>
        Consulter notre
         
        <span
          data-mockid="LinkComponent"
        >
          documentation partenaires
        </span>
      </p>
    </div>
  </div>
  <div
    class="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full"
  >
    <div
      class="fr-col-offset-1 fr-col-10"
    >
      <div
        data-mockid="DTO2FormComponent"
      />
    </div>
  </div>
</div>
`;

exports[`InstanceUpdatePage should call LinkComponent with params 1`] = `
<div>
  <h1
    class="fr-col-12 fr-col-md-8"
  >
    Partners-form-updateTitle-mock
  </h1>
  <div
    data-mockid="AlertComponent"
  >
    <div>
      AlertComponent
    </div>
    <div>
      <p>
        Consulter notre
         
        <span
          data-mockid="LinkComponent"
        >
          documentation partenaires
        </span>
      </p>
    </div>
  </div>
  <div
    class="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full"
  >
    <div
      class="fr-col-offset-1 fr-col-10"
    >
      <div
        data-mockid="DTO2FormComponent"
      />
    </div>
  </div>
</div>
`;

exports[`InstanceUpdatePage should match the snapshot 1`] = `
<div>
  <h1
    class="fr-col-12 fr-col-md-8"
  >
    Partners-form-updateTitle-mock
  </h1>
  <div
    data-mockid="AlertComponent"
  >
    <div>
      AlertComponent
    </div>
    <div>
      <p>
        Consulter notre
         
        <span
          data-mockid="LinkComponent"
        >
          documentation partenaires
        </span>
      </p>
    </div>
  </div>
  <div
    class="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full"
  >
    <div
      class="fr-col-offset-1 fr-col-10"
    >
      <div
        data-mockid="DTO2FormComponent"
      />
    </div>
  </div>
</div>
`;
